pkg/values/map.go

```diff
@@ -20,7 +20,7 @@ func EmptyMap() *Map {
 	}
 }
 
-func NewMap(m map[string]any) (*Map, error) {
+func NewMap[T any](m map[string]T) (*Map, error) {
```

Comment:
required all maps to be `map[string]any` now can also create values map from typed map (e.g., `map[string]string`)